src/main/java/com/company/spring_application/domain/Order.java

```diff
@@ -0,0 +1,75 @@
+package com.company.spring_application.domain;
+
+import java.io.Serializable;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
```

Comment:
This is mutable list, which is intended for very specific case when reads are often and writes are rare, it usually used for complex concurrency scenarios. For Order document you need just pure immutable collection of Product elements. It could be done with java collections api.